src/library/scala/collection/Iterator.scala

```diff
@@ -1259,7 +1259,10 @@ object Iterator extends IterableFactory[Iterator] {
         else adjustedBound min (until - lo)   // keep lesser bound
       if (rest == 0) empty
       else {
-        dropping += lo
+        dropping = {
+          val sum = dropping + lo
+          if (sum < 0) Int.MaxValue else sum
```

Comment:
Setting `dropping = Int.MaxValue` on overflow is not correct

```
scala> Iterator.from(0).drop(1).drop(Int.MaxValue).take(5).toList
val res3: List[Int] = List(2147483647, -2147483648, -2147483647, -2147483646, -2147483645)

scala> Iterator.from(0).drop(100).drop(Int.MaxValue).take(5).toList
val res4: List[Int] = List(2147483647, -2147483648, -2147483647, -2147483646, -2147483645)
```

Reply:
Though maybe that's OK? Generally iterators can have more than `Int.MaxValue` elements, but once we start using operations like `drop` that have an `Int` parameter, maybe it's too difficult / not necessary dealing with all overflow conditions?

Reply:
That's also what I thought. As our "size type" for all the collections in the std lib is always `Int`, I think this is fine. Imho everything below the size of `Int.MaxValue` has to work, but everything above is a bonus.